turbine/src/sigverify_shreds.rs

```diff
@@ -175,18 +218,64 @@ fn run_shred_sigverify<const K: usize>(
     Ok(())
 }
 
+#[must_use]
+fn verify_retransmitter_signature(
+    shred: &[u8],
+    root_bank: &Bank,
+    working_bank: &Bank,
+    cluster_info: &ClusterInfo,
+    leader_schedule_cache: &LeaderScheduleCache,
+    cluster_nodes_cache: &ClusterNodesCache<RetransmitStage>,
+    stats: &ShredSigVerifyStats,
+) -> bool {
+    let signature = match shred::layout::get_retransmiter_signature(shred) {
+        Ok(signature) => signature,
+        // If the shred is not of resigned variant,
+        // then there is nothing to verify.
+        Err(shred::Error::InvalidShredVariant) => return true,
```

Comment:
Seems like we'll need to require Resigned shred variant once we flip the switch. Will this then return false? Or should we reject shreds before they get here?

Reply:
What I am thinking right now is that:
* if a shred is of resigned variant, we verify if it was resigned by the right turbine parent.
* then once we have the whole block, we can verify that the last erasure batch was of the resigned variant before voting on that block (similar to verifying that it has 32+ data shreds).

The thing is that once we receive a shred we cannot say that if it belongs to the last erasure batch or not, until we receive a data shred with `LAST_SHRED_IN_SLOT`. So we cannot discard a shred here if it is not resigned (because we only require resigned variant for the last erasure batch).